src/Mod/Part/App/TopoShape.h

```diff
@@ -459,8 +529,74 @@ class PartExport TopoShape : public Data::ComplexGeoData
     mutable std::shared_ptr<TopoShapeCache> _cache;
     mutable TopLoc_Location _subLocation;
 
-private:
-    TopoDS_Shape _Shape;
+    class ShapeProtector : public TopoDS_Shape
+    {
+    public:
+        using TopoDS_Shape::TopoDS_Shape;
+        using TopoDS_Shape::operator=;
+
+        ShapeProtector() = default;
+        ShapeProtector(const TopoDS_Shape& shape) : TopoDS_Shape(shape) {}

```

Comment:
Is this not `explicit` intentionally? I can't recall if we need the automatic cast.

Reply:
I would say yes, this makes shapeprotector behave as if it was a topods_shape transparently.
Is this not right?